vaadin-app-layout-flow-parent/vaadin-app-layout-flow/src/main/java/com/vaadin/flow/component/applayout/AppLayout.java

```diff
@@ -51,6 +55,36 @@
 
     private Component content;
 
+    private AppLayoutI18n i18n;
+
+    /**
+     * Gets the internationalization object previously set for this component.
+     * <p>
+     * Note: updating the object content that is gotten from this method will
+     * not update the lang on the component if not set back using
+     * {@link AppLayout#setI18n(AppLayoutI18n)}
+     *
+     * @return the i18n object. It will be <code>null</code>, if the i18n
+     *         properties are not set.
+     */
+    public AppLayoutI18n getI18n() {
+        return i18n;
+    }
+
+    /**
+     * Sets the internationalization properties for this component.
+     *
+     * @param i18n
+     *            the internationalized properties, not <code>null</code>
+     */
+    public void setI18n(AppLayoutI18n i18n) {
```

Comment:
@sissbruecker I saw your https://github.com/vaadin/flow-components/pull/2067#discussion_r700826767 about `onAttach` behavior that also needs to be covered and I want to make sure if it makes practical sense for AppLayout.

I wonder if there is a use case where AppLayout could be detached and attached again? As far as I understand, it is typically used as a router layout that somewhat implies that it cannot be detached, am I correct?

Reply:
Discussed it internally: `onAttach` is only relevant when executing some JavaScript code with `executeJs` as this is not persisted by the Flow component over component's detach/attach.